docs/shadertrap_language.md

```diff
@@ -0,0 +1,338 @@
+    [ attribute_index_1 -> BUFFER vertex_buffer_1 OFFSET_BYTES offset_1 STRIDE_BYTES stride_1 DIMENSION dimension_1,
+      attribute_index_2 -> BUFFER vertex_buffer_2 OFFSET_BYTES offset_2 STRIDE_BYTES stride_2 DIMENSION dimension_2,
+      ...
+      attribute_index_n -> BUFFER vertex_buffer_n OFFSET_BYTES offset_n STRIDE_BYTES stride_n DIMENSION dimension_n,
+    ]
+  INDEX_DATA index_buffer
+  VERTEX_COUNT count
+  TOPOLOGY topology
+  FRAMEBUFFER_ATTACHMENTS
+    [ location_1 -> attachment_1,
+      location_2 -> attachment_2,
+      ...
+      location_n -> attachment_n
+    ]
+```
+
+Runs a graphics workload.
+
+- `graphics_program` must be a *graphics* program produced by `CREATE_PROGRAM`.
+- The `VERTEX_DATA` parameter is a mapping with one entry for each vertex attribute associated with the vertex shader of `graphics_program`. If the vertex shader declares an `in` variable with `layout(location = i)` then `VERTEX_DATA` must provide an entry of the form `i -> ...`.  In this entry:
+  - `vertex_buffer_i` specifies a buffer from which vertex data for vertex attribute `i` will be taken. This must be produced by `CREATE_BUFFER`. The elements of this buffer will be interpreted as 32-bit floating-point numbers - i.e. the `GL_FLOAT` type is used for vertex data.
+  - `offset_i` is a non-negative integer specifying the byte offset into `vertex_buffer_i` at which vertex data begins
+  - `stride_i` is a non-negative integer specifying the distance in bytes between successive pieces of vertex data in `vertex_buffer_i`
+  - `dimension_i` is a positive integer specifying the dimensionality of vertices associated with vertex attribute `i`
+- `index_buffer` specifies a buffer of index data, which must be produced by `CREATE_BUFFER`. It will be interpreted as a buffer of unsigned 32-bit integers - i.e. the `GL_UNSIGNED_INT` type is used for index data.
+- `count` is a non-negative integer specifying how many vertices should be processed. Thus `index_buffer` should contain at least this many unsigned integers, and for each unsigned integer `index` in `index_buffer` the vertex buffers associated with each vertex attribute should contain suitable data.
+- `topology` specifies the kind of primitive to be drawn using the vertex data. At present only `TRIANGLES` is supported. [More kinds of primitive should be supported](https://github.com/google/shadertrap/issues/25).
+- For every `out` variable in the fragment shader associated with `graphics_program` there should be a corresponding entry in the `FRAMEBUFFER_ATTACHMENTS` parameter. If the fragment shader has a declaration of the form `out layout(location = l)` then `FRAMEBUFFER_ATTACHMENTS` should have an entry `location_i -> attachment_i` such that `location_i` = `l`, and `attachment_i` is a renderbuffer produced by `CREATE_RENDERBUFFER` or a texture produced by `CREATE_EMPTY_TEXTURE_2D`. This supports off-screen rendering to both renderbuffers and textures. On-screen rendering is not supported.
+
```

Comment:
An example could be helpful to understand the syntax of that command with the mapping.

Reply:
I have added a TODO for adding an example in the future. I agree that it would be useful.